tests/lightning_qubit/test_measurements_class.py

```diff
@@ -465,19 +476,40 @@ def test_single_return_value(self, measurement, observable, lightning_sv, tol):
             if isinstance(observable, list)
             else [measurement(op=observable)]
         )
-        tape = qml.tape.QuantumScript(ops, measurements)
+        tape = qml.tape.QuantumScript(ops, measurements, shots=shots)
 
-        expected = self.calculate_reference(tape, lightning_sv)
         statevector = lightning_sv(n_qubits)
         statevector = statevector.get_final_state(tape)
         m = LightningMeasurements(statevector)
-        result = m.measure_final_state(tape)
+
+        skip_list = (
+            qml.ops.Sum,
+            # qml.Hamiltonian,
+            qml.SparseHamiltonian,
+        )
+        do_skip = measurement is qml.var and isinstance(observable, skip_list)
+        do_skip = do_skip or (
+            measurement is qml.expval and isinstance(observable, qml.SparseHamiltonian)
+        )
+        do_skip = do_skip and shots is not None
+        if do_skip:
+            with pytest.raises(TypeError):
+                _ = m.measure_final_state(tape)
+            return
+        else:
+            result = m.measure_final_state(tape)
+
+        expected = self.calculate_reference(tape, lightning_sv)
 
         # a few tests may fail in single precision, and hence we increase the tolerance
-        assert np.allclose(result, expected, max(tol, 1.0e-4))
+        if shots is None:
+            assert np.allclose(result, expected, max(tol, 1.0e-4))
+        else:
```

Comment:
Could we just relax the tol for single precision calculation?

Reply:
I already checked it but it is not possible to reduce the number of shots and precision at the same time without test fails. What do you suggest, reduce the shots or increase the precision? :thinking:

Reply:
You can annotate the test as [flaky](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/tests/lightning_qubit/test_measurements_class.py#L421), and Pytest will re-run it if it fails to see if it works.

Reply:
It is usually best to avoid it, but sometimes this is the only "fast" solution.

Reply:
I checked it and this test already [has](https://github.com/PennyLaneAI/pennylane-lightning/pull/810/files#r1727201473) the `flaky` option  :confused: 

We can increase the `flaky`'s number but I am not sure if this works :thinking: